source/shared/cpp/ObjectModel/TextBlock.cpp

```diff
@@ -165,3 +168,126 @@ void TextBlock::SetHorizontalAlignment(const HorizontalAlignment value)
     m_hAlignment = value;
 }
 
+std::string TextBlock::ParseISO8601UsingRegex() const
+{
+	std::regex pattern("\\{\\{(DATE|TIME)\\((\\d{4})-{1}(\\d{2})-{1}(\\d{2})T(\\d{2}):{1}(\\d{2}):{1}(\\d{2})(Z|([+-])(\\d{2}):{1}(\\d{2}))(\\)|, S[Hh][Oo][Rr][Tt]\\)|, L[Oo][Nn][Gg]\\))\\}\\}");
+    std::smatch matches;
+    std::string text = m_text;
+    std::ostringstream parsedostr;
+    time_t offset = 0; 
+    bool isDate = false, isShort = true;
+    enum MatchIndex
+    {
+        IsDate = 1,
+        YEAR,
+        Month,
+        Day,
+        Hour,
+        Min,
+        Sec,
+        TimeZone,
+		TZHr = 10,
+		TZMn,
+        Format,
+    };
+    int factor = 1, hr = 0, mn = 0;
```

Comment:
>int factor = 1, hr = 0, mn = 0; [](start = 4, length = 31)

This code is a little tricky to read.  Could you add some more comments and maybe longer/more descriptive variable names?  (ie, what does "factor" represent?  "mn" is minutes?  Lets just call it minutes)

Reply:
sure; thanks for insights


---
In reply to: [149151448](https://github.com/Microsoft/AdaptiveCards/pull/871#discussion_r149151448) [](ancestors = 149151448)